sql/core/src/main/java/org/apache/spark/sql/vectorized/ArrowColumnVector.java

```diff
@@ -433,10 +418,14 @@ final long getLong(int rowId) {
   private static class ArrayAccessor extends ArrowVectorAccessor {
 
     private final ListVector accessor;
+    private final ArrowBuf offsets;
+    private final ArrowColumnVector arrayData;
 
     ArrayAccessor(ListVector vector) {
       super(vector);
       this.accessor = vector;
+      this.offsets = vector.getOffsetBuffer();
```

Comment:
Maybe `vector.getOffsetBuffer()` will change when loading the next batch so we can't reuse the `ArrowBuf`.

Reply:
good catch!